File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    private static int sum(String group) {
        int sum = 0;
        for (String num : group.split("\\+")) {
            sum += Integer.parseInt(num);
        }

        return sum;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] groups = br.readLine().split("-");

        int result = sum(groups[0]);
        for (int i = 1; i < groups.length; i++) {
            result -= sum(groups[i]);
        }

        System.out.println(result);
    }
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun sumNumbers(group: String): Int {
    return group.split("+").sumOf { it.toInt() }
}

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val groups = br.readLine().split("-")

    var result = sumNumbers(groups[0])
    for (i in 1 until groups.size) {
        result -= sumNumbers(groups[i])
    }

    println(result)
}